core/src/main/java/bio/terra/pearl/core/service/search/terms/SearchValue.java

```diff
@@ -83,38 +83,57 @@ public static SearchValue ofNestedProperty(Object object, String field, SearchVa
         }
     }
 
+    private Object getValue() {
+        return switch (this.searchValueType) {
+            case STRING -> this.stringValue;
+            case NUMBER -> this.numberValue;
+            case INSTANT -> this.instantValue;
+            case DATE -> this.dateValue;
+            case BOOLEAN -> this.booleanValue;
+            case ARRAY -> this.arrayValue;
+            default -> null;
+        };
+    }
+
     public boolean equals(SearchValue right) {
         if (!this.searchValueType.equals(SearchValueType.ARRAY) && right.searchValueType.equals(SearchValueType.ARRAY)) {
```

Comment:
This wasn't from this PR but this strikes me as a bit too clever and makes the below code seem incomplete.  I'd recommend two helper functions "isUseArrayCompare()" and "arrayEquals"